### PR TITLE
AMRVAC: variable He abundance

### PR DESCRIPTION
**Pull request by jordidj**
## PR Summary

Removed hardcoded He abundance for AMRVACDataset unit calculation. Instead, it is now read from the AMRVAC parfile.

## PR Checklist

- [ ] New features are documented, with docstrings and narrative docs
- [ ] Adds a test for any bugs fixed. Adds tests for new features.

### CODE REVIEW

File: yt/frontends/amrvac/data_structures.py [hunk 1]
```diff
@@ -373,10 +374,14 @@ def _set_code_unit_attributes(self):
...
         mp_cgs = self.quan(1.672621898e-24, "g")  # This value is taken from AstroPy
-        He_abundance = 0.1  # hardcoded parameter in AMRVAC
 
         # get self.length_unit if overrides are supplied, otherwise use default
         length_unit = getattr(self, "length_unit", self.quan(1, "cm"))
+        namelist = read_amrvac_namelist(self._parfiles)
+        if "mhd_list" in namelist:
+            He_abundance = namelist["mhd_list"].get("he_abundance", 0.1)
+        else:
+            He_abundance = 0.1  # default
```
Comment (neutrinoceros) [inline yt/frontends/amrvac/data_structures.py:384]: Inlining allows to keep a single source of truth for the default value
```suggestion
        He_abundance = namelist.get("mhd_list", {}).get("he_abundance", 0.1)
```

File: yt/frontends/amrvac/data_structures.py [hunk 2]
```diff
@@ -203,20 +203,8 @@ def __init__(
...
 
         self._geometry_override = geometry_override
-        super().__init__(
-            filename,
-            dataset_type,
-            units_override=units_override,
-            unit_system=unit_system,
-            default_species_fields=default_species_fields,
-        )
+        self._parfiles = None
```
Comment (neutrinoceros) [inline yt/frontends/amrvac/data_structures.py:206]: using an empty list here allows some simplifications in the rest of your patch
```suggestion
        self._parfiles = []
```

File: yt/frontends/amrvac/data_structures.py [hunk 3]
```diff
@@ -373,10 +376,14 @@ def _set_code_unit_attributes(self):
...
         mp_cgs = self.quan(1.672621898e-24, "g")  # This value is taken from AstroPy
-        He_abundance = 0.1  # hardcoded parameter in AMRVAC
 
         # get self.length_unit if overrides are supplied, otherwise use default
         length_unit = getattr(self, "length_unit", self.quan(1, "cm"))
+        if self._parfiles is not None:
+            namelist = read_amrvac_namelist(self._parfiles)
+            He_abundance = namelist.get("mhd_list", {}).get("he_abundance", 0.1)
+        else:
+            He_abundance = 0.1
```
Comment (neutrinoceros) [inline yt/frontends/amrvac/data_structures.py:386]: We can still inline it with my previous suggestion
```suggestion
     namelist = read_amrvac_namelist(self._parfiles)
     He_abundance = namelist.get("mhd_list", {}).get("he_abundance", 0.1)
```

Comment (jordidj) [reply yt/frontends/amrvac/data_structures.py:386]: `read_amrvac_namelist` currently doesn't work on empty lists as you suggest, so we need a check either here or in `io.py`. Do you have a preference?

**Comment by neutrinoceros** [comment]
please make sure `pre-commit` linting stays green. Right now it looks like your latest push introduced a SyntaxError

**Review by neutrinoceros — APPROVED** [review]
Thank you ! I'll merge when test pass.

**Comment by chrishavlin** [comment]
Thanks for the PR @jordidj !

**Pull request merged**